File: android/src/main/kotlin/dev/note11/flutter_naver_map/flutter_naver_map/util/location/NDefaultMyLocationTrackerHandler.kt
package dev.note11.flutter_naver_map.flutter_naver_map.util.location

import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel

internal interface NDefaultMyLocationTrackerHandler {
    fun handle(call: MethodCall, result: MethodChannel.Result) = when (call.method) {
        "requestLocationPermission" -> requestLocationPermission(result)
        "getCurrentPositionOnce" -> getCurrentPositionOnce(result)
        "cancelAllWaitingGetCurrentPositionOnceTask" -> cancelAllWaitingGetCurrentPositionOnceTask()
        else -> result.notImplemented()
    }

    fun requestLocationPermission(result: MethodChannel.Result)

    fun getCurrentPositionOnce(result: MethodChannel.Result)

    fun cancelAllWaitingGetCurrentPositionOnceTask()
}

File: ios/flutter_naver_map/Sources/flutter_naver_map/util/location/NDefaultMyLocationTrackerHandler.swift
import Foundation
import Flutter

internal protocol NDefaultMyLocationTrackerHandler {
    func requestLocationPermission(result: @escaping FlutterResult)

    func getCurrentPositionOnce(result: @escaping FlutterResult)
}

internal extension NDefaultMyLocationTrackerHandler {
    func handle(call: FlutterMethodCall, result: @escaping FlutterResult) {
        switch (call.method) {
        case "requestLocationPermission":
            requestLocationPermission(result: result)
        case "getCurrentPositionOnce":
            getCurrentPositionOnce(result: result)
        default:
            result(FlutterMethodNotImplemented)
        }
    }
}
